bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Pattern.java

```diff
@@ -39,22 +41,19 @@
  */
 public class Pattern extends Resource {
 
-	/**
-	 * the OS resource for the Pattern
-	 * (Warning: This field is platform dependent)
-	 * <p>
-	 * <b>IMPORTANT:</b> This field is <em>not</em> part of the SWT
-	 * public API. It is marked public only so that it can be shared
-	 * within the packages provided by SWT. It is not available on all
-	 * platforms and should never be accessed from application code.
-	 * </p>
-	 *
-	 * @noreference This field is not intended to be referenced by clients.
-	 */
-	public long handle;
+	private int initialZoom;
 
 	private Runnable bitmapDestructor;
 
+	private Image image;
+
+	private float x1, y1, x2, y2;
+	private Color color1, color2;
+	private int alpha1, alpha2;
+	private final boolean isImagePattern;
```

Comment:
Add some comments / JavaDocs to explain what these all are. The names are not descriptive. Even better: rename to something more descriptive so you don't even need the JavaDocs.

Reply:
Added